test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -317,7 +317,9 @@ g.test_old_leader_txn_during_promote_write = function(g)
     --
     -- Server2 gets the bad txn.
     --
-    g.server1:wait_for_downstream_to(g.server2)
+    t.helpers.retrying({}, function()
+        g.server2:assert_follows_upstream(g.server1:get_instance_id())
+    end)
```

Comment:
Why change this?

Reply:
`wait_for_downstream_to` simply waits until there's a corresponding relay.
We want to wait until `server2` really gets the txn, that is until it is synced.
Since this commit changes `replication_sync_timeout` to 0, `box.cfg{replication=...}` exits before nodes are synced.
That's why `assert_follows_upstream` is needed now: it waits until server2's upstream is in "follow" state.

Reply:
Probably `wait_for_downstream_to` wasn't even needed before, when each replication reconfiguration returned only after sync